core/trino-main/src/main/java/io/trino/dispatcher/LocalDispatchQuery.java

```diff
@@ -123,8 +123,8 @@ private void waitForMinimumWorkers()
             }
             ListenableFuture<Void> minimumWorkerFuture = clusterSizeMonitor.waitForMinimumWorkers(executionMinCount, getRequiredWorkersMaxWait(session));
             // when worker requirement is met, start the execution
-            addSuccessCallback(minimumWorkerFuture, () -> startExecution(queryExecution));
-            addExceptionCallback(minimumWorkerFuture, throwable -> queryExecutor.execute(() -> stateMachine.transitionToFailed(throwable)));
+            addSuccessCallback(minimumWorkerFuture, () -> startExecution(queryExecution), queryExecutor);
+            addExceptionCallback(minimumWorkerFuture, throwable -> stateMachine.transitionToFailed(throwable), queryExecutor);
```

Comment:
These seem like unrelated changes.  This moves the call back from a direct executor to an explicit executor, which may be useful for some case, but that seems to be unrelated to what is the core of this commit.

Reply:
Right. This is not directly related to this PR. I thought it would be small enough to be included in this PR as a small clean-up. However, if you think that it would be better to have a separate PR, I will exclude this change from this PR. Please let me know.